client/my-sites/email/inbox/new-mailbox-upsell/index.jsx

```diff
@@ -0,0 +1,57 @@
+import { Button } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import { useSelector } from 'react-redux';
+import { hasGSuiteWithUs } from 'calypso/lib/gsuite';
+import { hasTitanMailWithUs } from 'calypso/lib/titan';
+import { recordInboxNewMailboxUpsellClickEvent } from 'calypso/my-sites/email/email-management/home/utils';
+import { INBOX_SOURCE } from 'calypso/my-sites/email/inbox/constants';
+import { emailManagement, emailManagementEdit } from 'calypso/my-sites/email/paths';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+
+/**
+ * Import styles
+ */
+import './style.scss';
+
+const NewMailboxUpsell = ( { domains } ) => {
+	const translate = useTranslate();
+	const selectedSite = useSelector( getSelectedSite );
+	const selectedSiteSlug = selectedSite?.slug;
+
+	let upsellURL = emailManagement( selectedSiteSlug, null, null, { source: INBOX_SOURCE } );
+
+	// User has one single domain, determine subscribed email provider.
+	if ( domains.length === 1 ) {
+		if ( hasTitanMailWithUs( domains[ 0 ] ) ) {
+			upsellURL = emailManagementEdit( selectedSiteSlug, domains[ 0 ].domain, 'titan/new', null, {
+				source: INBOX_SOURCE,
+			} );
+		} else if ( hasGSuiteWithUs( domains[ 0 ] ) ) {
+			upsellURL = emailManagementEdit(
+				selectedSiteSlug,
+				domains[ 0 ].domain,
+				'google-workspace/add-users',
+				null,
+				{ source: INBOX_SOURCE }
+			);
+		}
+	}
```

Comment:
```suggestion
	const getSectionSlug = ( domain ) => {
		if ( hasTitanMailWithUs( domain ) ) {
			return 'titan/new';
		}

		if ( hasGSuiteWithUs( domain ) ) {
			return 'google-workspace/add-users';
		}	
		
		return null
	};

	if ( domains.length === 1 && getSectionSlug( domain[ 0 ] ) ) {
		upsellURL = emailManagementEdit( selectedSiteSlug, domains[ 0 ].domain, getSectionSlug( domain [ 0 ] ), null, {
				source: INBOX_SOURCE,
			} );
	}
```

Nit: would this be more readable? Less logic duplication?

Reply:
I like the idea of calculating a `slug` and removed the duplicated `emailManagementEdit()` :) I've adapted a similar approach, but without introducing a new helper function `getSectionSlug()`. I'd like to avoid having to call it twice, when we could have just stored the calculated value in a variable.